### PR TITLE
When building Arrow shared look for a shared OpenSSL

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
When asked to build arrow from source, state that we want a shared library version of OpenSSL instead of a static. That ensures that we use the OpenSSL provided by conda ( which only provides shared ).

Plus in general building against a static OpenSSL is a bad idea as it makes deploying SSL CVE fixes very hard. 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/cmake/thirdparty/get_arrow.cmake [hunk 1]
```diff
@@ -126,9 +126,14 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
     # Turn off CPM using `find_package` so we always download and make sure we get proper static
     # library.
     set(CPM_DOWNLOAD_Arrow TRUE)
+    # We don't want to explicitly force a static OpenSSL when `BUILD_STATIC` so we allow the
```
Comment (bdice) [inline cpp/cmake/thirdparty/get_arrow.cmake:129]: This feels like it’s phrased a little backwards. Maybe explain the default behavior and how users can change it. The comment below this one follows that approach and makes more sense for a first-time reader in my opinion. For example:

By default ARROW will try to search for a static version of OpenSSL which is a bad idea given that shared linking is advised for critical components like SSL. If a static build is requested, we honor ARROW's default of static linking, but users may consider setting `ARROW_OPENSSL_USE_SHARED` even in static builds.

**Review by bdice — APPROVED** [review]
LGTM with a proposed comment edit

**Comment by bdice** [comment]
Clarifying: did you mean “shared”? From the context I expected the other word.

> That ensures that we use the OpenSSL provided by conda ( which only provides **static** ).

**Comment by robertmaynard** [comment]
> Clarifying: did you mean “shared”? From the context I expected the other word.
> 
> > That ensures that we use the OpenSSL provided by conda ( which only provides **static** ).

Whoops, you are correct. Fixed.

**Review by vyasr — APPROVED** [review]

**Comment by vyasr** [comment]
@gpucibot merge

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**